### ISSUE TITLE
support HTTP/3

### ISSUE THREAD

**Issue by xquery** [curl/curl-container#23]
https://github.com/curl/curl-docker/issues/76

**Comment by xquery** [comment]
will target an initial experimental release eg. curl-http3-exp which will use hyper as backend

**Comment by xquery** [comment]
now that curl has removed expiremental - time to do this here

**Comment by marcelofernandez** [comment]
Hi!

I forked this repo to get a working image of curl with HTTP/3 support so that I could set up an [HTTP/3 Kathara laboratory](https://github.com/redesunlu/kathara-labs/tree/main/kathara-lab_http3) for our Network Fundamentals University course:

https://github.com/redesunlu/curl-container-http3

Although I only did the changes needed to get a debian-dev curl working image, you could merge those changes in a new branch here as a starting point. I could add some help to see this implemented upstream (given enough of my spare time).

Regards

**Comment by akostadinov** [comment]
any updates?

**Comment by xquery** [comment]
yes, we want to do - wanted the openssl vs gnutls debate to calm down first
